docs/topics/virtual-warping.rst

```diff
@@ -54,3 +54,116 @@ extract pixels corresponding to its central zoom 9 tile, do the following.
+        -ts 200 250 \
+        tests/data/RGB.byte.tif \
+        tests/data/WGS84-RGB.byte.tif
+
+So, the attributes of these two images drastically differ:
+
+.. code-block:: console
+
+    $ rio info --shape tests/data/RGB.byte.tif
+    718 791
+    $ rio info --shape tests/data/WGS84-RGB.byte.tif
+    250 200
+    $ rio info --crs tests/data/RGB.byte.tif
+    EPSG:32618
+    $ rio info --crs tests/data/WGS84-RGB.byte.tif
+    EPSG:4326
+    $ rio bounds --bbox --geographic --precision 7 tests/data/RGB.byte.tif
+    [-78.95865, 23.5649912, -76.5749237, 25.5508738]
+    $ rio bounds --bbox --geographic --precision 7 tests/data/WGS84-RGB.byte.tif
+    [-78.9147773, 24.119606, -76.5963819, 25.3192311]
+
+and this snippet demonstrates how to normalize data to consistent dimensions,
+CRS, and cell size within a pre-defined bounding box:
+
+.. code-block:: python
+
+    from __future__ import division
+
+    import os
+
+    import affine
+
+    import rasterio
+    from rasterio.crs import CRS
+    from rasterio.enums import Resampling
+    from rasterio import shutil as rio_shutil
+    from rasterio.vrt import WarpedVRT
+
+
+    input_files = (
+        # This file is in EPSG:32618
+        'tests/data/RGB.byte.tif',
+        # This file is in EPSG:4326
+        'tests/data/WGS84-RGB.byte.tif'
+    )
+
+    # Destination CRS is Web Mercator
+    dst_crs = CRS.from_epsg(3857)
+
+    # These coordiantes are in Web Mercator
+    dst_bounds = -8744355, 2768114, -8559167, 2908677
+
+    # Output image dimensions
+    dst_height = dst_width = 100
+
+    # Output image transform
+    left, bottom, right, top = dst_bounds
+    xres = (right - left) / dst_width
+    yres = (top - bottom) / dst_height
+    dst_transform = affine.Affine(xres, 0.0, left,
+                                  0.0, -yres, top)
+
+    vrt_options = {
+        'resampling': Resampling.cubic,
+        'dst_crs': dst_crs,
+        'dst_transform': dst_transform,
+        'dst_height': dst_height,
+        'dst_width': dst_width,
+    }
+
+    for path in input_files:
+
+        with rasterio.open(path) as src:
+
+            with WarpedVRT(src, **vrt_options) as vrt:
+
+                # At this point 'vrt' is a full dataset with dimensions,
+                # CRS, and spatial extent matching 'vrt_options'.
+
+                # Read all data into memory.
+                data = vrt.read()
+
+                # Process the dataset in chunks.  Likely not very efficient.
+                for _, window in vrt.block_windows():
+                    data = vrt.read(window=window)
+
+                # Dump the aligned data into a new file.  A VRT representing
+                # this transformation can also be produced by switching
+                # to the VRT driver.
+                directory, name = os.path.split(path)
+                outfile = os.path.join(directory, 'aligned-{}'.format(name))
+                rio_shutil.copy(vrt, outfile, driver='GTiff')
```

Comment:
@cmarshak and @sebhahn this `rasterio.shutil.copy()` will produce a new dataset without a CRS unless you are running the fix included in this PR.  It can be replaced with:

```
meta = vrt.meta.copy()
meta['driver'] = 'GTiff'
with rasterio.open(outfile, 'w', **meta) as dst:
    dst.write(vrt.read())
```

Or if your image is larger than available memory (this is likely to be slow):

```
meta = vrt.meta.copy()
meta['driver'] = 'GTiff'
with rasterio.open(outfile, 'w', **meta) as dst:
    for _, window in vrt.block_windows():
        data = vrt.read(window=window)
        dst.write(data, window=window)
```

Reply:
@geowurster Thanks for the example!